src/network/networkd-sysctl.c

```diff
@@ -40,10 +42,69 @@ static void manager_set_ip_forwarding(Manager *manager, int family) {
                                   enable_disable(t), af_to_ipv4_ipv6(family));
 }
 
+static int sysctl_inotify(sd_event_source *source, const struct inotify_event *event, void *userdata) {
+        _cleanup_free_ char *buf = NULL, *path = NULL;
+        Hashmap *sysctl_shadow = ASSERT_PTR(userdata);
+        const char *dir_path, *value;
+        int r;
+
+        assert(event);
+
+        r = sd_event_source_get_inotify_path(source, &dir_path);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to get the watched inotify path: %m");
+
+        path = path_join(dir_path, event->name);
+        if (!path)
+                return log_oom_debug();
+
+        value = hashmap_get(sysctl_shadow, path);
+
+        /* We don't handle this sysctl */
+        if (!value)
+                return r;
+
+        r = read_virtual_file(path, SIZE_MAX, &buf, NULL);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to read sysctl value '%s': %m", path);
+
+        delete_trailing_chars(buf, "\n");
+
+        if (!streq(value, buf))
+                log_warning("sysctl %s was changed from '%s' to '%s'", path, value, buf);
+
+        return 0;
+}
+
+static int manager_inotify_add(Manager *manager) {
```

Comment:
maybe call this manager_sysctl_inotify_add(), just to match then name of the handler

Reply:
so this function here you named manager_inotify_add() but the per-link one you called link_watch_sysctl(). I think they should have a more similar name, since they do very similar things. Hence maybe:

manager_watch_sysctl() and link_watch_sysctl()?

or

manager_inotify_sysctl_add() and link_inotify_sysctl_add()?

I think i prefer the former, but you pick, i just care that this is somewhat systematic, and contains "sysctl" in the name.